indico/htdocs/js/indico/widgets/multiple_items_widget.js

```diff
@@ -187,46 +191,70 @@
 
         function makeColData(item, col, forceEditable) {
             if (item && !forceEditable) {
-                return col.type === 'select' ? {text: options.columnChoices[col.id][item[col.id]]} : {text: item[col.id]};
-            } else {
                 if (col.type === 'select') {
-                    var sel = $('<select>', {
-                        'class': 'js-table-input',
-                        'data-required': col.required
-                    });
-                    sel.append($('<option>')); // Default empty option
-                    for (var choiceID in options.columnChoices[col.id]) {
-                        sel.append($('<option>', {
-                            value: choiceID,
-                            text: options.columnChoices[col.id][choiceID],
-                            selected: item ? item[col.id] === choiceID : false
-                        }));
+                    return {text: options.columnChoices[col.id][item[col.id]]};
+                } else if (col.type === 'checkbox') {
+                    // return {text: item[col.id] ? '✓' : ''};
+                    return {
+                        html: $('<span>', {
+                            class: 'icon-checkbox-{0}'.format(item[col.id] ? 'checked' : 'unchecked')
+                        }),
+                        css: {
+                            'text-align': 'center'
+                        }
+                    };
+                } else {
+                    return {text: item[col.id]};
+                }
+            } else if (col.type === 'select') {
+                var sel = $('<select>', {
+                    'class': 'js-table-input',
+                    'data-required': col.required
+                });
+                sel.append($('<option>')); // Default empty option
+                for (var choiceID in options.columnChoices[col.id]) {
+                    sel.append($('<option>', {
+                        value: choiceID,
+                        text: options.columnChoices[col.id][choiceID],
+                        selected: item ? item[col.id] === choiceID : false
+                    }));
+                }
+                return {html: sel};
+            } else if (col.type === 'checkbox') {
+                return {
+                    html: $('<input>', {
+                        type: 'checkbox',
+                        class: 'js-table-input',
+                        value: '1',
+                        checked: item ? item[col.id] : false
+                    }),
+                    css: {
+                        width: '1%'
```

Comment:
When the dict has only 1 element, why not having it inline (like you do in line `222`)?